src/lib/libhtml5.js

```diff
@@ -805,6 +807,59 @@ var LibraryHTML5 = {
   emscripten_set_focusout_callback_on_thread: (target, userData, useCapture, callbackfunc, targetThread) =>
     registerFocusEventCallback(target, userData, useCapture, callbackfunc, {{{ cDefs.EMSCRIPTEN_EVENT_FOCUSOUT }}}, "focusout", targetThread),
 
+  $registerInputEventCallback__proxy: 'sync',
+  $registerInputEventCallback__deps: ['$JSEvents', '$findEventTarget', '$stringToUTF8', '$lengthBytesUTF8', 'malloc', 'realloc'],
+  $registerInputEventCallback: (target, userData, useCapture, callbackfunc, eventTypeId, eventTypeString, targetThread) => {
+#if PTHREADS
+    targetThread = JSEvents.getTargetThreadForEventCallback(targetThread);
+#endif
+
+    var inputEventHandlerFunc = (e = event) => {
+        var data = e.data || '';
+        var inputType = e.inputType || '';
+        var isComposing = e.isComposing;
+        const dataLengthWithTermination = lengthBytesUTF8(data) + 1;
+        const eventSize = {{{ C_STRUCTS.EmscriptenInputEvent.__size__ }}} + dataLengthWithTermination;
+        if(JSEvents.inputEvent == null){
+            JSEvents.inputEvent = _malloc(eventSize);
+            JSEvents.inputEventAllocatedSize = eventSize;
+        }
+        else if(JSEvents.inputEventAllocatedSize < eventSize) {
+            JSEvents.inputEvent = _realloc(JSEvents.inputEvent, eventSize);
+            JSEvents.inputEventAllocatedSize = eventSize;
```

Comment:
Maybe just call `realloc` unconditionally and then you don't need to trace `inputEventAllocatedSize` at all.